### PR TITLE
added mqtt module

### PR DESCRIPTION
**Pull request by Towflos**
added a module to publish mavlink messages to mqtt

### CODE REVIEW

File: MAVProxy/modules/mavproxy_mqtt.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        elif args[0] == 'connect':
+            self.connect()
+
+    def usage(self):
+        """show help on command line options"""
+        return "Usage: mqtt <set|connect>"
+
+    def convert_to_dict(self, message):
+        """converts mavlink message to python dict"""
+        if hasattr(message, '_fieldnames'):
```
Comment (peterbarker) [inline MAVProxy/modules/mavproxy_mqtt.py:58]: when is this not the case?

**Review by peterbarker — APPROVED** [review]
This looks interesting.

I'm going to merge this - it's not going to break anyone who doesn't load it.

**Comment by peterbarker** [comment]
Merged, thanks!

File: MAVProxy/modules/mavproxy_mqtt.py (hunk 1 above)
Comment (Towflos) [reply MAVProxy/modules/mavproxy_mqtt.py:58]: I was not sure if there could be messages that contain another message as field that's why the function is recursive. This checks if the function gets called on a field with a message or a value.

**Comment by peterbarker** [comment]
On Thu, 31 Mar 2022, Towflos wrote:

> I was not sure if there could be messages that contain another message as field that's why the function is recursive. This checks if the function gets called on a field with a message or a value.

Yeah, that doesn't happen :-)

Feel free to simplify.

**Pull request merged**